cpp/src/arrow/filesystem/azurefs.h

```diff
@@ -21,6 +21,7 @@
 #include <string>
 #include <vector>
 
+#include "arrow/filesystem/azurefs_internal.h"
```

Comment:
Is this a mistake? We should not include an internal header in a public header.

Reply:
It's a mistake! Fixing.